tdishr/TdiTrans.c

```diff
@@ -291,8 +291,15 @@ int Tdi1Trans(int opcode, int narg, mdsdsc_t *list[], mdsdsc_xd_t *out_ptr)
       psig->dimensions[dim] = 0;
     pmask = (mdsdsc_t *)&ncopies;
     /** scalar to simple vector **/
-    if (rank == 0)
-      memcpy((char *)&arr, (char *)pa, head);
+    if (rank == 0) {
+      memcpy((char *)&arr, (char *)pa, sizeof(struct descriptor));
+      arr.dimct = 1;
+      arr.aflags.coeff = 0;
+      arr.a0 = arr.pointer;
+      arr.arsize = arr.length;
+      arr.m[1] = arr.m[0] = 1;
```

Comment:
Style-wise, we should avoid multiple assignments on a single line, it can get hard to read.